### DISCUSSION TITLE
Missing charset in Content-Type

### DISCUSSION THREAD

**Discussion by pbryan** [aio-libs/aiohttp#7187] [Q&A] [answered]
I've encountered an issue with Salesforce's bulk query API: it returns data with a Content-Type of `text/csv`, but without a `charset` parameter. According to RFC 2616, the behavior is for the user-agent to interpret it as ISO-8859-1 encoded text. Which is what aiohttp seems to do when I use `response.text()`. However, RFC 9110 makes no mention of how to interpret text without the `charset` parameter, and states (in a quasi-related section) that "…UTF-8 has become nearly ubiquitous…".

So, I'm wondering:

1. Is there a current RFC that addresses this issue of missing character set parameter in `text/*`?
2. Can anyone suggest how I might best solve this with aiohttp? Barring any better suggestion, it looks like I'll need to use `response.read()` and decode the body myself as UTF-8.

**Comment by Dreamsorcerer** [comment, marked as answer]
Well, the documentation clearly shows an encoding parameter:
https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.text

I assume that's all you need as a workaround.

Whether we should change the default to utf-8 or not, I'm not sure (it does sound reasonable to me).

**Reply by pbryan** [reply]
I missed the encoding param, thanks.
